src/couch_replicator_httpc_pool.erl

```diff
@@ -55,114 +58,113 @@ release_worker(Pool, Worker) ->
 
 init({Url, Options}) ->
     process_flag(trap_exit, true),
-    State = #state{
+    State0 = case couch_replicator_dns:lookup(Url) of
```

Comment:
Don't know if relevant but if the lookup hangs, could that be for long? And would it matter that it blocks the gen_server start up sequence possible for that long? Not sure if that could lead to the server seeming to hang during start up.